extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -28,46 +28,62 @@ public ThreadContextSnapshot currentContext(Map<String, String> map) {
 
         // capture the state, null indicates no active context while capturing snapshot
         InjectableContext.ContextState state = isContextActiveOnThisThread(arc) ? arc.requestContext().getState() : null;
-        return () -> {
-            // can be called later on, we should retrieve the container again
-            ArcContainer arcContainer = Arc.container();
-            if (arcContainer == null || !arcContainer.isRunning()) {
-                throw new IllegalStateException("Arc context propagation was attempted but the container is not running.");
```

Comment:
Most changes are from removing lambdas, the real change is removing the exception and just making it a noop.

Reply:
Sounds good for shutdown, but what if this happens in other scenarios? Shouldn't we at least log something?

Reply:
I don't think it matters, if the container is gone there is no request context anymore.

Reply:
Actually. never mind. Checking on history, this check was added specifically because of shutdown scenario.